rfd/0225-teleport-extended-support.md

```diff
@@ -0,0 +1,163 @@
+number of release branches and build pipelines. 
+
+### Extended Support
+
+While customers might be asking for an LTS release explicitly, what they really want are two
+things: stability and extended support. We can achieve these without introducing a Teleport LTS
+release by changing our process slightly.
+
+We can reduce the frequency of Teleport major releases from every ~4 months to once a year and cut
+down the number of supported releases from N-2 to N-1. That would extend the support window for major
+releases from ~12 months to ~24 months at the expense of supporting fewer major versions. Reducing
+the number of versions also means it takes fewer upgrade cycles to get to the most recent version of Teleport.
+No changes will be made to the upgrade guide. Users may upgrade their cluster from any N-1 version to any
+N version.
+
+This satisfies customer desire for major releases to be supported for a longer period of time
+without impacting our current upgrade process or component compatibility guarantees. However, it
+comes at the expense of reducing the frequency at which we can make breaking changes.
+
+As dictated by semantic versioning, a major release should be reserved for making any backward-incompatible
+changes to the public API. We've often relied on this four month release cycle to quickly iterate
+on APIs that don't work or cause issues for one reason or another. With yearly major releases this
+becomes a less viable solution and a stronger emphasis must be put both on backward and forward compatibility. 
+
+Examples of forward compatibility issues that we've seen in the past:
+
+- Using a boolean instead of a string or an enum - https://protobuf.dev/best-practices/dos-donts/#bad-bools
+- Strict validation over a closed set of values prevents adding new ones
+- Making new fields mandatory
+
+Examples of backward compatibility issues that we've seen in the past:
+- Removing a deprecated API too soon
+- Changing the behavior of an existing API
+- Altering validation of existing fields of a resource
+
+Going forward, solutions to problems should attempt to preserve compatibility at all costs and
+rely less on all customers adhering to our upgrade procedure. Instead of relying on versions of
+clients, and that the Auth server will reject outdated clients we should put a stronger emphasis
+on negotiating compatibility. If clients and servers in our control can inform each other about
+what features they support in some manner then both can mutually agree on which features and APIs
+can be safely used. An example of this can be seen with the capabilities exchanged during instance
+heartbeat hello messages: https://github.com/gravitational/teleport/blob/6d4961aed1eef6b360e05facdbcd997fb95593d4/api/proto/teleport/legacy/client/proto/inventory.proto#L136-L181.
+
+To address the stability concerns, all new features *MUST* only be added to the most recent major version.
```

Comment:
It would be useful to document explicitly what we consider acceptable to go into a release that's in maintenance mode. For example:

- Security fixes.
- Regression fixes.
- Bug fixes that affect customer deployments (unless it's non-trivial?)
- Critical performance fixes that affect customer deployments (unless it's non-trivial?)

I think it will be hard to draw a boundary and we'd probably have to decide on a case by case basis but some starting guidance might be useful.